### PR TITLE
Add some details about view compilation options

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
https://github.com/aspnet/MvcPrecompilation/issues/62 is covered by #99 and other existing tests. Instead using #62 to detail some of the options that allow configuring the tool.

File: README.md [hunk 1]
```diff
@@ -6,4 +6,40 @@ Travis:   [![Travis](https://travis-ci.org/aspnet/MvcPrecompilation.svg?branch=d
 
 The Razor syntax provides a fast, terse, clean, and lightweight way to combine server code with HTML to create dynamic web content. This repo contains tooling that allows compilation of MVC Razor views as part of build and publish.
 
+## Installation and usage
+Add a package reference to `Microsoft.AspNetCore.Mvc.Razor.ViewCompilation` in your project:
+
+```xml
+<ItemGroup>
+  <Microsoft.AspNetCore.Mvc.Razor.ViewCompilation Version="1.0.0" />
```
Comment (Eilon) [inline README.md]: This... doesn't look right?

Comment (Eilon) [inline README.md]: And why 1.0?

File: README.md [hunk 2]
```diff
@@ -6,4 +6,40 @@ Travis:   [![Travis](https://travis-ci.org/aspnet/MvcPrecompilation.svg?branch=d
 
 The Razor syntax provides a fast, terse, clean, and lightweight way to combine server code with HTML to create dynamic web content. This repo contains tooling that allows compilation of MVC Razor views as part of build and publish.
 
+## Installation and usage
+Add a package reference to `Microsoft.AspNetCore.Mvc.Razor.ViewCompilation` in your project:
```
Comment (Eilon) [inline README.md]: Doesn't the metapackage handle this for people?

File: README.md [hunk 3]
```diff
@@ -6,4 +6,40 @@ Travis:   [![Travis](https://travis-ci.org/aspnet/MvcPrecompilation.svg?branch=d
...
+    Name="PrecompileRazorViews" 
+    AfterTargets="Build"
+    DependsOnTargets="MvcRazorPrecompile" />
+```
+
+## Options
+
+Some aspects of view compilation can be configured by editing the project:
+
+* `MvcRazorExcludeViewFilesFromPublish `: Enabling `MvcRazorCompileOnPublish` prevents .cshtml files from being published. This option disables this behavior.
```
Comment (Eilon) [inline README.md]: 😮 trailing space before one of the ` marks

File: README.md [hunk 4]
```diff
@@ -6,4 +6,40 @@ Travis:   [![Travis](https://travis-ci.org/aspnet/MvcPrecompilation.svg?branch=d
...
+    AfterTargets="Build"
+    DependsOnTargets="MvcRazorPrecompile" />
+```
+
+## Options
+
+Some aspects of view compilation can be configured by editing the project:
+
+* `MvcRazorExcludeViewFilesFromPublish `: Enabling `MvcRazorCompileOnPublish` prevents .cshtml files from being published. This option disables this behavior. 
+Note: Mvc Core does not support updateable precompiled views. Any modifications to published cshtml files will be ignored if a precompiled view is discovered for that path.
```
Comment (Eilon) [inline README.md]: It's called "ASP.NET Core MVC"

File: README.md (hunk 2 above)
Comment (pranavkm) [inline README.md]: Not on desktop. I guess I can say if you aren't referencing the .All package, use this.

File: README.md (hunk 1 above)
Comment (pranavkm) [inline README.md]: We don't have a 2.0.0 🎱

File: README.md [hunk 5]
```diff
@@ -6,4 +6,44 @@ Travis:   [![Travis](https://travis-ci.org/aspnet/MvcPrecompilation.svg?branch=d
 
 The Razor syntax provides a fast, terse, clean, and lightweight way to combine server code with HTML to create dynamic web content. This repo contains tooling that allows compilation of MVC Razor views as part of build and publish.
 
+## Installation and usage
+
+### Referencing the `Microsoft.AspNetCore.Mvc.Razor.ViewCompilation` package
+* If you're targeting ASP.NET Core 2.0 or higher on CoreCLR, a reference to the `Microsoft.AspNetCore.Mvc.Razor.ViewCompilation` package is added by `Microsoft.AspNetCore.All` and you do not need to explicitly reference it.
```
Comment (rynowak) [inline README.md]: `CoreCLR` isn't really right - it's `netcoreapp2.0`

File: README.md [hunk 6]
```diff
@@ -6,4 +6,44 @@ Travis:   [![Travis](https://travis-ci.org/aspnet/MvcPrecompilation.svg?branch=d
...
+* For desktop targeting projects or projects targeting ASP.NET Core 1.x, add a package reference to the appropriate version of `Microsoft.AspNetCore.Mvc.Razor.ViewCompilation` in your project:
+
+```xml
+<ItemGroup>
+  <Microsoft.AspNetCore.Mvc.Razor.ViewCompilation Version="1.1.1" />
```
Comment (rynowak) [inline README.md]: Isn't this `<PackageReference .. />?

File: README.md [hunk 7]
```diff
@@ -6,4 +6,44 @@ Travis:   [![Travis](https://travis-ci.org/aspnet/MvcPrecompilation.svg?branch=d
...
+</ItemGroup>
+```
+
+### Enabling view compilation
+To enable compilation as part of publishing, add the `MvcRazorCompileOnPublish` property:
+```xml
+<PropertyGroup>
+  <MvcRazorCompileOnPublish>true</MvcRazorCompileOnPublish>
+</PropertyGroup>
+```
```
Comment (rynowak) [inline README.md]: Is this still required or is it on by default?

Comment (pranavkm) [inline README.md]: Yeah it is now. I'll add a note saying you need this if you aren't referencing the WebSdk

**Comment by pranavkm** [comment]
🆙 📅

**Pull request merged**